### ISSUE TITLE
Diffeomorphic Registration using SyNRA

### ISSUE THREAD

**Issue by asmaharry** [ANTsX/ANTsPy#185]
Diffeomorphic Registration using SyNRA with MI and reg_iterations=(40,20,0), on an image size of 240 240 155 took only 3 minutes, I am confused how this diffeomorphic registration is too fast? or is there any other parameters that i have to set.
please suggest me a good transformation type for diffeomorphic registration and the other parameters.

**Comment by stnava** [comment]
did you try the `verbose` option to see what is going on?

three minutes is about right if you have a decent computer.   the published information about timing is largely either out of date or related to the `CC` metric.

**Comment by asmaharry** [comment]
I also tried SyN with MI and with CC. with CC it took almost 1 hour and 40 minutes but with MI it took 3 minutes. the difference between the two time elapses is very large. 
if i use the type_of_transform='SyN' then it is deformable registration, am i right?

**Comment by stnava** [comment]
yes -  'CC' can take between 10 minutes and 10 hours (or more) depending on the other parameters you choose

**Comment by asmaharry** [comment]
if i use the verbose, it gives me metric value and convergence value. what will be the good convergence value, like initially it gives me inf and then some numbers, tell me how to evaluate the registration on the basis of these values?

**Comment by stnava** [comment]
lower is better.

brian

**Comment by asmaharry** [comment]
Thank you for the reply.
is the metric Value in 3rd column a mutual information score?
how the convergence value is computed? what is it?

**Comment by ntustison** [comment]
>> is the metric Value in 3rd column a mutual information score?

Yes.

>> how the convergence value is computed? what is it?

Depending on the specified "window size" (default = 10), the the previous n="window size" metric values are used to compute a line.  The slope of that line is the convergence value at that iteration.
